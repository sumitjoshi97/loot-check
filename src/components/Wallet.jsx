import React, { Component } from 'react'
import { connect } from 'react-redux'
import { deposit, withdraw } from '../actions/balance'

export class Wallet extends Component {
  state = {
    balance: 0
  }

  updateBalance = e => {
    this.setState({
      balance: parseInt(e.target.value, 10)
    })
  }
  render() {
    return (
      <div>
        <h3 className="balance">Wallet Balance: {this.props.balance}</h3>
        <input className="input-wallet" onChange={this.updateBalance} />
        <button
          className="btn-deposit"
          onClick={() => this.props.deposit(this.state.balance)}
        >
          Deposit
        </button>
        <button
          className="btn-withdraw"
          onClick={() => this.props.withdraw(this.state.balance)}
        >
          Withdrawl
        </button>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  balance: state.balance.balance
})

export default connect(
  mapStateToProps,
  { deposit, withdraw }
)(Wallet)
