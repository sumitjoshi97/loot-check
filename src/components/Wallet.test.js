import React from 'react'
import { shallow } from 'enzyme'
import { Wallet } from './Wallet'

describe('Wallet', () => {
  const props = { balance: 0 }
  const wallet = shallow(<Wallet {...props} />)

  it('renders component successfully', () => {
    expect(wallet).toMatchSnapshot()
  })

  it('displays balance from props', () => {
    expect(wallet.find('.balance').text()).toEqual('Wallet Balance: 0')
  })
})
