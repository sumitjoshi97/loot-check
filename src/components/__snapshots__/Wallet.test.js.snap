// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3
        className="balance"
      >
        Wallet Balance: 
        0
      </h3>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Wallet Balance: ",
          0,
        ],
        "className": "balance",
      },
      "ref": null,
      "rendered": Array [
        "Wallet Balance: ",
        0,
      ],
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3
          className="balance"
        >
          Wallet Balance: 
          0
        </h3>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet Balance: ",
            0,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Wallet Balance: ",
          0,
        ],
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
